### PR TITLE
[terra-form-radio] Null check for child component so that it won't cr…

### PR DESCRIPTION
**Pull request by yfaney**
…ash if it's null

### Summary
I added the null check for child components so that it won't crash if it's null.

Please add your name to the [CONTRIBUTORS.md] file. Adding your name to the [CONTRIBUTORS.md] file signifies agreement to all rights and reservations provided by the [License].

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-form-radio/src/RadioField.jsx [hunk 1]
```diff
@@ -123,7 +123,7 @@ const RadioField = (props, { intl }) => {
   );
 
   const content = React.Children.map(children, (child) => {
-    if (child.type.isRadio) {
+    if (child && child.type.isRadio) {
```
Comment (tbiethman) [inline packages/terra-form-radio/src/RadioField.jsx]: Can you include a Jest test for this scenario?

Comment (yfaney) [reply packages/terra-form-radio/src/RadioField.jsx]: Added: b53561c

File: packages/terra-form-radio/package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "classnames": "^2.2.5",
     "lodash.uniqueid": "^4.0.1",
     "prop-types": "^15.5.8",
+    "raf-polyfill": "^1.0.0",
```
Comment (StephenEsser) [inline packages/terra-form-radio/package.json]: Why was this added?

Comment (yfaney) [reply packages/terra-form-radio/package.json]: No idea. Removed: https://github.com/cerner/terra-core/pull/2314/commits/6374930ee51190515c85b4726ae6514a15f42c67#diff-afd97ad7b73244896cf178257a620ee2L31

**Comment by yfaney** [comment]
Added changelog as per a bot check: https://github.com/cerner/terra-core/pull/2314/commits/6374930ee51190515c85b4726ae6514a15f42c67#diff-2acab889c3500ac761bbaa2fd7a19b4fR6

File: packages/terra-form-radio/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,8 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
+* Adding a null check to the child components to avoid a crash
```
Comment (yuderekyu) [inline packages/terra-form-radio/CHANGELOG.md]: nitpick - your change protects against all falsey values. Not just null.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 4]
```diff
@@ -9,6 +9,13 @@ it('should render a default radio field', () => {
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render a default radio field without crashing if it has a null child', () => {
```
Comment (yuderekyu) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx]: either change this to 
```suggestion
it('should render a default radio field without crashing if it has an undefined child', () => {
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 5]
```diff
@@ -9,6 +9,13 @@ it('should render a default radio field', () => {
...
+  const nullChild = undefined;
```
Comment (yuderekyu) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx]: or
```suggestion
  const nullChild = null;
```

File: packages/terra-form-radio/CHANGELOG.md (hunk 3 above)
Comment (yfaney) [reply packages/terra-form-radio/CHANGELOG.md]: Updated: 0bf88b3a0

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 4 above)
Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: Updated: efc1d732b

File: packages/terra-form-radio/src/RadioField.jsx (hunk 1 above)
Comment (yfaney) [reply packages/terra-form-radio/src/RadioField.jsx]: @tbiethman I tried to run jest to create the snapshot but it gives me an error. What do I need to do to fix this?

```
● Validation Error:

  Module raf/polyfill in the setupFiles option was not found.
         <rootDir> is: /Users/yj048444/git/terra-core

  Configuration Documentation:
  https://jestjs.io/docs/configuration.html
```

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 5 above)
Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: I picked undefined: efc1d73

File: packages/terra-form-radio/src/RadioField.jsx (hunk 1 above)
Comment (tbiethman) [reply packages/terra-form-radio/src/RadioField.jsx]: raf is a dependency of terra-toolkit: https://github.com/cerner/terra-core/blob/master/package.json#L91. Can you verify that those dependencies are present in your node_modules directory?

Comment (yfaney) [reply packages/terra-form-radio/src/RadioField.jsx]: I found the cause. I needed to run compile first LOL
Updated: d95158091

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 6]
```diff
@@ -9,6 +9,13 @@ it('should render a default radio field', () => {
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render a default radio field without crashing if it has an undefined child', () => {
+  const undefinedChild = undefined;
+  const radioField = (<RadioField legend="Default RadioField">{ undefinedChild }</RadioField>);
```
Comment (emilyrohrbough) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx]: Should this example just be
```<RadioField legend="Default RadioField" />```
or are we actually wanting to prevent  passing undefined as the children value?

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx [hunk 7]
```diff
@@ -9,6 +9,13 @@ it('should render a default radio field', () => {
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render a default radio field without crashing if it has an undefined child', () => {
```
Comment (emilyrohrbough) [inline packages/terra-form-radio/tests/jest/RadioField.test.jsx]: nit: lets remove without crashing. all of these tests shouldn't crash the component 😃

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 6 above)
Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: Prevent from passing undefined as the children value. In other words, we are testing the RadioField component with a child component which is undefined.

File: packages/terra-form-radio/tests/jest/RadioField.test.jsx (hunk 7 above)
Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: Removed: 56b2ee965

Comment (danplubell) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: There is also a potential for crashing if the child does not have the type property.

Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: @danplubell K I will cover that as well.

Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: @danplubell Just curious, is that possible?

Comment (danplubell) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: Yes, if the child is not a Radio element or anything else that might not have the type property.  For instance, there isn't anything to prevent someone from having a mix of Radio elements and Select elements as children of the RadioField.

Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: Added: 0f1b14558

Comment (mjhenkes) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: @danplubell & @yfaney I discussed this with the terra team and we're ok allowing non-radio elements in radio form field. Namely the case we discussed was adding some kind of image next to a radio button. Radio form field will add screen reader accessibility to direct radio children, any non radio children will need to be evaluated for accessibility. We don't recommend adding any other form elements to the radio field but should not prevent it.

Comment (yfaney) [reply packages/terra-form-radio/tests/jest/RadioField.test.jsx]: @mjhenkes @danplubell Reverted: 192dab011

File: packages/terra-form-radio/src/RadioField.jsx (hunk 1 above)
Comment (bjankord) [inline packages/terra-form-radio/src/RadioField.jsx]: We should update [checkboxfield](https://github.com/cerner/terra-core/blob/master/packages/terra-form-checkbox/src/CheckboxField.jsx#L126) to work the same as well. @yfaney can you update that as part of this PR as well.

Comment (yfaney) [reply packages/terra-form-radio/src/RadioField.jsx]: @bjankord Added: 9b8324a7a
